pkg/stringset/set.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package stringset
+
+import (
+	cuckoo "github.com/seiflotfy/cuckoofilter"
+)
+
+type Set interface {
+	Has(string) bool
+	HasAny([]string) bool
+}
+
+type fixedSet struct {
+	cuckoo *cuckoo.Filter
+}
+
+func (f fixedSet) HasAny(strings []string) bool {
+	for _, s := range strings {
+		if f.Has(s) {
+			return true
+		}
+	}
+	return false
+}
+
+func NewFromStrings(items ...string) Set {
+	f := cuckoo.NewFilter(uint(len(items)))
```

Comment:
Maybe we could estimate the size of the underlying slices as per the comments here https://github.com/seiflotfy/cuckoofilter/blob/master/cuckoofilter.go#L21-L26 and add some way of adding an upper limit for this, something like maybe `5MB` by default?

Reply:
Hm, what action do we take if the limit is exceeded?

Reply:
I wonder if that is a good tradeoff though, because 5MB is a fairly low price to pay compared to the increase in memory required for buffering and resorting series. If `1000000` is ~1MB, I think it will be very hard to have so many label names for memory of the filter to be a problem. Maybe we should check behavior in production before we add limits?

Reply:
:+1: makes sense